src/testdrive/src/format/avro.rs

```diff
@@ -260,45 +261,50 @@ where
-                        .replace_all(&format!("{:#?}", a).to_string(), regex_replacement.as_str())
-                        .to_string(),
-                    _ => format!("{:#?}", a),
-                };
+    if !debug_print_only {
+        let mut found_beginning = !partial_search;
+        let mut expected_item = expected.next();
+        let mut actual_item = actual.next();
+        loop {
+            let i = index.next().expect("known to exist");
+            match (expected_item, actual_item) {
+                (Some(e), Some(a)) => {
+                    let e_str = format!("{:#?}", e);
+                    let a_str = match &regex {
+                        Some(regex) => regex
+                            .replace_all(
+                                &format!("{:#?}", a).to_string(),
+                                regex_replacement.as_str(),
+                            )
+                            .to_string(),
+                        _ => format!("{:#?}", a),
+                    };
 
-                if e_str != a_str {
-                    if found_beginning {
-                        bail!(
-                            "record {} did not match\nexpected:\n{}\n\nactual:\n{}",
-                            i,
-                            e_str,
-                            a_str
-                        );
+                    if e_str != a_str {
+                        if found_beginning {
+                            bail!(
+                                "record {} did not match\nexpected:\n{}\n\nactual:\n{}",
+                                i,
+                                e_str,
+                                a_str
+                            );
+                        }
+                        actual_item = actual.next();
+                    } else {
+                        found_beginning = true;
+                        expected_item = expected.next();
+                        actual_item = actual.next();
                     }
-                    actual_item = actual.next();
-                } else {
-                    found_beginning = true;
-                    expected_item = expected.next();
-                    actual_item = actual.next();
                 }
-            }
-            (Some(e), None) => bail!("missing record {}: {:#?}", i, e),
-            (None, Some(a)) => {
-                if !partial_search {
-                    bail!("extra record {}: {:#?}", i, a);
+                (Some(e), None) => bail!("missing record {}: {:#?}", i, e),
+                (None, Some(a)) => {
+                    if !partial_search {
+                        bail!("extra record {}: {:#?}", i, a);
+                    }
+                    break;
                 }
-                break;
+                (None, None) => break,
             }
-            (None, None) => break,
         }
     }
     let expected: Vec<_> = expected.map(|e| format!("{:#?}", e)).collect();
```

Comment:
do we want to print expected if `debug_print_only` is true? or just `actual?

Reply:
whoops yeah bad cherry pick! but now that I'm looking at it again, I can simplify